app/lib/ui/widgets/utility/pdf_screen.dart

```diff
@@ -71,33 +89,128 @@ class _PDFScreenState extends State<PDFScreen> {
+              child: pw.Text("Shipper's Information",
+                  style: pw.TextStyle(fontSize: 18))),
+          pw.Paragraph(
+              text: widget.atr.shipInfo.toString(),
+              style: pw.TextStyle(fontSize: 16)),
+          pw.Header(
+              level: 1,
+              child: pw.Text("Transporter's Information",
+                  style: pw.TextStyle(fontSize: 18))),
+          pw.Paragraph(
+              text: widget.atr.tranInfo.toString(),
+              style: pw.TextStyle(fontSize: 16)),
+          pw.Header(
+              level: 1,
+              child: pw.Text("Feed Water and Rest Information",
+                  style: pw.TextStyle(fontSize: 18))),
+          pw.Paragraph(
+              text: widget.atr.fwrInfo.toString(),
+              style: pw.TextStyle(fontSize: 16)),
+          pw.Header(
+              level: 1,
+              child: pw.Text("Loading Vehicle Information",
+                  style: pw.TextStyle(fontSize: 18))),
+          pw.Paragraph(
+              text: widget.atr.vehicleInfo.toString(),
+              style: pw.TextStyle(fontSize: 16)),
+          pw.Header(
+              level: 1,
+              child: pw.Text("Delivery Information",
+                  style: pw.TextStyle(fontSize: 18))),
+          pw.Paragraph(
+              text: widget.atr.deliveryInfo.toString(),
+              style: pw.TextStyle(fontSize: 16)),
+          pw.Header(
+              level: 1,
+              child: pw.Text("Acknowledgements",
+                  style: pw.TextStyle(fontSize: 18))),
+          pw.Text("Transporter's Acknowledgment"),
+          pw.SizedBox(
+            height: 250,
+            child: pw.Image(tranImage),
+          ),
+          pw.Padding(padding: pw.EdgeInsets.all(20)),
+          pw.Text("Shipper's Acknowledgment"),
+          pw.SizedBox(
+            height: 250,
+            child: pw.Image(shipperImage),
+          ),
+          pw.Padding(padding: pw.EdgeInsets.all(20)),
+          pw.Text("Receiver's Acknowledgment"),
+          pw.SizedBox(
+            height: 250,
+            child: pw.Image(receiverImage),
+          ),
+          pw.Padding(padding: pw.EdgeInsets.all(25)),
+          pw.Header(
+              level: 1,
+              child: pw.Text("Contingency Plan",
+                  style: pw.TextStyle(fontSize: 18))),
+          pw.Paragraph(
+              text: widget.atr.contingencyInfo.toString(),
+              style: pw.TextStyle(fontSize: 16)),
         ];
       },
     ));
 
-    // TODO: Change the filename to something useful
-    return Future.wait([getTemporaryDirectory(), pdf.save()]).then(
-        (List values) =>
-            File("${values[0].path}/example.pdf").writeAsBytes(values[1]));
+    return Future.wait(
+        [getTemporaryDirectory(), pdf.save()]).then((List values) => File(
+            "${values[0].path}/${atr.deliveryInfo.recInfo.destinationLocationName}${atr.deliveryInfo.arrivalDateAndTime}.pdf")
```

Comment:
Even though all our PDFs are in the temp directory, there is a possibility the names could collide.

- [ ] You could add the ATR document ID in the name again to ensure unique: `${atr.deliveryInfo.recInfo.destinationLocationName}${atr.deliveryInfo.arrivalDateAndTime}${atr.identifier.documentId}`